cfpb/consumerfinance.gov
# Add missing Ask search label

Adds a visually hidden text label for the Ask search input when that input has no other label. Note that the input will have a label if `show_label` is true—the `h2` or `h3` that appears above the input—hence all the ugly Jinja conditionals to prevent two labels in that case. (Is a `span` the best way to mark up the search icon when another label is there?)

Fixes GHE/CFGOV/platform/issues/4149.

---




## Additions

- Visually hidden label for the Ask search input if that input has no other text label


## Changes

- For some reason we were using the raw SVG instead of the `svg_icon` in the template tag, so I swapped that out—does that have any impacts I overlooked?

## How to test this PR

1. Run a WAVE check on https://www.consumerfinance.gov/complaint/ and note the two errors related to the Ask search input: "Missing form label" and "Empty form label". Then run a WAVE check on https://www.consumerfinance.gov/ask-cfpb/ and note the "Empty form label" error.
2. Run the same WAVE checks on this branch and note the errors related to the Ask search input are gone.
3. Check to make sure that the magnifying glass icon is marked up as a `label` and has a visually hidden text description when `show_label` is false (as on the complaint landing page) but marked up as a `span` with no text description when `show_label` is true (as on the Ask landing page or any Ask question page).

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
